apmproxy/receiver.go

```diff
@@ -107,6 +107,15 @@ func (c *Client) handleInfoRequest() (func(w http.ResponseWriter, r *http.Reques
 		r.Header.Set("X-Forwarded-Host", r.Header.Get("Host"))
 		r.Host = parsedApmServerURL.Host
 
+		// Send authorization header if the APM agent did not already forwarded it
+		if r.Header.Get("Authorization") == "" {
```

Comment:
Instead of checking if the `Authorization` header is set or not can we check if the secret manager is configured or not and override the headers if it is? In my mind, this seems to be a better approach than adding Auth headers if not provided but I might be mistaken entirely.

Reply:
I'm not sure about that approach.

This was adapted from what we are doing in the forwarder:

https://github.com/elastic/apm-aws-lambda/blob/bf8fc3fafb7ccb87f9a0e0c9cf97991a248f4010/apmproxy/apmserver.go#L163-L167

The client has not knowledge of aws  or the secrets manager. It only knows about the apikey and the secret token.

WDYT ?

Reply:
> This was adapted from what we are doing in the forwarder:

IIUC, the forwarder is different than the handle info request. The info request is a pure reverse proxy whereas the forwarder creates a new request and thus always adds the auth headers. I am a bit concerned about `if r.Header.Get("Authorization") == ""` condition; what do you think about overriding the auth headers unconditionally?

Reply:
good point! I updated the code!